src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -608,42 +614,68 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    .rs.fromJSON(.rs.base64decode(encoded))
 })
 
+.rs.addFunction("parseYamlOpt", function(opt)
```

Comment:
FWIW, we have `.Call("rs_fromYAML")` if you need to parse a YAML string into an R object. Would that be helpful here?

I think you could just trim off the `#| ` prefix and then use that to parse the whole thing together.

Reply:
Yeah, so based on the documentation you sent, the string could be in two different format styles, R and YAML



```
#| tag = Rvalue, tag = Rvalue, tag = Rvalue
```
or
```
#| tag: value
#| tag: value
```

Reply:
Right, but _hopefully_ users will prefer to use YAML rather than R code here. I don't know if it's reasonable to try-catch with a YAML parser first and fall back to an R parser.